### PR TITLE
Support minimization and maximization of metrics

### PR DESCRIPTION
**Pull request by rgemulla**
via a new config key `valid.metric_max`

@Daniel: This is largely untested & I may also have missed some places where metrics are compared.

### CODE REVIEW

**Comment by rufex2001** [comment]
@rgemulla the metric.py file was not added, so I can't test this yet

**Comment by rgemulla** [comment]
Added.

File: kge/util/metric.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+from typing import Union, Iterable, List
+
+
+class Metric:
+    "Utility class for comparing metrics."
+
+    def __init__(self, metric_max: Union[Job, Config, bool]):
+        "Params: metric_max=True means higher is better."
+        if isinstance(metric_max, Job):
+            metric_max = metric_max.config
```
Comment (rufex2001) [inline kge/util/metric.py:12]: Isn't the .get("valid.metric_max") missing here?

File: kge/util/optimizer.py [hunk 2]
```diff
@@ -32,6 +32,24 @@ def __init__(self, config: Config, optimizer):
...
+            # check for consistency of metric-based scheduler
+            self._metric_based = name in ["ReduceLROnPlateau"]
+            if self._metric_based:
+                desired_mode = "max" if config.get("valid.metric_max") else "min"
+                if "mode" in args:
+                    if args["mode"] != desired_mode:
+                        raise ValueError(
+                            (
+                                "valid.metric_max ({}) and train.lr_scheduler_args.mode "
+                                "({}) are inconsistent."
```
Comment (rufex2001) [inline kge/util/optimizer.py:44]: Nice! This has to match indeed!

**Review by rufex2001 — COMMENTED** [review]
All looks good and also works well. Tested it with a couple of runs, also tested early stopping. Just line 12 in metric.py seems off, the rest is fine!

File: kge/util/metric.py (hunk 1 above)
Comment (rgemulla) [reply kge/util/metric.py:12]: The line right below does this.

**Comment by rufex2001** [comment]
Right! I automatically read those lines as mutually exclusive...
